lib/lightning_web/live/project_live/new_collaborator_component.ex

```diff
@@ -0,0 +1,65 @@
+defmodule LightningWeb.ProjectLive.NewCollaboratorComponent do
+  @moduledoc false
+
+  use LightningWeb, :live_component
+
+  alias Lightning.Accounts.UserNotifier
+  alias Lightning.Projects
+  alias LightningWeb.ProjectLive.Collaborators
+  alias Phoenix.LiveView.JS
+
+  @impl true
+  def update(assigns, socket) do
+    collaborators = %Collaborators{}
+
+    changeset = Collaborators.changeset(collaborators, %{})
+
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign(collaborators: collaborators, changeset: changeset)}
+  end
+
+  @impl true
+  def handle_event("validate", %{"project" => params}, socket) do
+    changeset =
+      Collaborators.changeset(socket.assigns.collaborators, params)
+
+    {:noreply, assign(socket, changeset: changeset)}
+  end
+
+  def handle_event(
+        "add_collaborators",
+        %{"project" => params},
+        %{assigns: assigns} = socket
+      ) do
+    with {:ok, project_users} <-
```

Comment:
We need to check if the user is allowed to call this function, even if the modal is closed this event provides an unverified opening to alter project users.

Reply:
Ah right good point. 👍